src/main/java/ui/components/DrawnCard.java

```diff
@@ -37,7 +39,7 @@ public Rank getRank() {
         return rank;
     }
 
-    public void getRank(Rank rank) {
+    public void setRank(Rank rank) {
         this.rank = rank;
```

Comment:
This class overall could use some light documentation just for readability purposes.